content/docs/04-ai-sdk-ui/03-chatbot-message-persistence.mdx

```diff
@@ -325,3 +325,165 @@ When the client reloads the page after a disconnect, the chat will be restored f
   the case where the client reloads the page after a disconnection, but the
   streaming is not yet complete.
 </Note>
+
+## Resuming ongoing streams
+
+<Note>This feature is experimental and may change in future versions.</Note>
+
+The `useChat` hook has experimental support for resuming an ongoing chat generation stream by any client, either after a network disconnect or by reloading the chat page. This can be useful for building applications that involve long-running conversations or for ensuring that messages are not lost in case of network failures.
+
+The following are the pre-requisities for your chat application to support resumable streams:
+
+- Installing the [`resumable-stream`](https://www.npmjs.com/package/resumable-stream) package that helps create and manage the publisher/subscriber mechanism of the streams.
+- Creating a [Redis](https://vercel.com/marketplace/redis) instance to store the stream state.
+- Creating a table that tracks the stream IDs associated with a chat.
+
+To resume a chat stream, you will use the `experimental_resume` function returned by the `useChat` hook. You will call this function during the initial mount of the hook inside the main chat component.
+
+```tsx filename="app/components/chat.tsx"
+'use client'
+
+import { useChat } from "@ai-sdk/react";
+import { Input } from "@/components/input";
+import { Messages } from "@/components/messages";
+
+export function Chat() {
+  const { experimental_resume } = useChat({id});
+
+  useEffect(() => {
+    experimental_resume();
+
+    // we use an empty dependency array to
+    // ensure this effect runs only once
+  }, [])
+
+  return (
+    <div>
+      <Messages>
+      <Input/>
+    </div>
+  )
+}
+```
+
+The `experimental_resume` function makes a `GET` request to your configured chat endpoint (or `/api/chat` by default) whenever your client calls it. If there’s an active stream, it will pick up where it left off, otherwise it simply finishes without error.
+
+The `GET` request automatically appends the `chatId` query parameter to the URL to help identify the chat the request belongs to. Using the `chatId`, you can look up the most recent stream ID from the database and resume the stream.
+
+```bash
+GET /api/chat?chatId=<your-chat-id>
+```
+
+Earlier, you must've implemented the `POST` handler for the `/api/chat` route to create new chat generations. When using `experimental_resume`, you must also implement the `GET` handler for `/api/chat` route to resume streams.
+
+### 1. Implement the GET handler
+
+Add a `GET` method to `/api/chat` that:
+
+1. Reads `chatId` from the query string
+2. Validates it’s present
+3. Loads any stored stream IDs for that chat
+4. Returns the latest one to `streamContext.resumableStream()`
+5. Falls back to an empty stream if it’s already closed
+
+```ts filename="app/api/chat/route.ts"
```

Comment:
The highlighting is broken - wonder if this will lead to issues with the docs rendering. Did you check it in the docs deployment?

Reply:
Hmm not sure why github isn't highlighting the code snippet, fwiw I did try building this in the docs deployment and it works!